common/src/main/java/tachyon/ClientBase.java

```diff
@@ -188,4 +191,90 @@ protected synchronized InetSocketAddress getAddress() {
     return mAddress;
   }
 
+  /**
+   * The RPC to be executed in {@link #retryRPC(RpcCallable)}.
+   *
+   * @param <V> the return value of {@link #call()}
+   */
+  protected interface RpcCallable<V> {
+    /**
+     * The task where RPC happens.
+     *
+     * @return RPC result
+     * @throws TException when any exception defined in thrift happens
+     */
+    V call() throws TException;
+  }
+
+  /**
+   * Same with {@link RpcCallable} except that this RPC call throws {@link TachyonTException} and
+   * is to be executed in {@link #retryRPC(RpcCallableThrowsTachyonTException)}.
+   *
+   * @param <V> the return value of {@link #call()}
+   */
+  protected interface RpcCallableThrowsTachyonTException<V> {
```

Comment:
Instead of hard coding the exception type into the interface name, do:

```
protected interface RpcCallable<T,U> {
  T call() throws U, TException;
}
```

and 

```
protected <T,U> T retryRPC(RpcCallable<T,U> rpc) throws U, IOException {
  ...
}
```